assets/js/common/ApiKeySettingsModal/ApiKeySettingsModal.test.jsx

```diff
@@ -0,0 +1,245 @@
+import React from 'react';
+import { render, screen, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { faker } from '@faker-js/faker';
+import {
+  addMonths,
+  getMonth,
+  getYear,
+  getDay,
+  addYears,
+  addDays,
+} from 'date-fns';
+import { truncate } from 'lodash';
+import ApiKeySettingsModal from './ApiKeySettingsModal';
+import '@testing-library/jest-dom';
+
+describe('ApiKeySettingsModal', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('Generation form', () => {
+    it('render the generation form', async () => {
+      const user = userEvent.setup();
+
+      await act(async () => {
+        render(<ApiKeySettingsModal open />);
+      });
+
+      expect(screen.getByText('Never Expires')).toBeVisible();
+      expect(screen.getByText('Key Expiration')).toBeVisible();
+      expect(screen.getByText('Key Expiration')).toBeVisible();
+      expect(
+        screen.getByText(
+          'By generating a new key, you will need to replace the API key on all hosts.'
+        )
+      ).toBeVisible();
+      expect(screen.getByRole('spinbutton')).toBeVisible();
+      expect(screen.getByRole('button', { name: 'Generate' })).toBeVisible();
+      expect(screen.getByRole('button', { name: 'Close' })).toBeVisible();
+      expect(screen.getByRole('button', { name: 'months' })).toBeVisible();
+
+      await user.click(screen.getByRole('button', { name: 'months' }));
+
+      expect(screen.getByRole('listbox')).toBeVisible();
+      expect(screen.getAllByRole('option')).toHaveLength(3);
+      expect(screen.getByRole('option', { name: 'years' })).toBeVisible();
+
+      expect(screen.getByRole('option', { name: 'months' })).toBeVisible();
+
+      expect(screen.getByRole('option', { name: 'days' })).toBeVisible();
+    });
+
+    it('should show a validation error when quantity is 0 and the form is enabled', async () => {
+      const user = userEvent.setup();
+
+      await act(async () => {
+        render(<ApiKeySettingsModal open />);
+      });
+
+      await user.type(screen.getByRole('spinbutton'), '0');
+
+      await user.click(screen.getByRole('button', { name: 'Generate' }));
+
+      expect(
+        screen.getByText('Key expiration value needs to be greater than 0')
+      ).toBeVisible();
+    });
+
+    it('should not show a validation error when quantity is > 0 and the form is enabled', async () => {
+      const user = userEvent.setup();
+
+      await act(async () => {
+        render(<ApiKeySettingsModal open />);
+      });
+
+      await user.type(screen.getByRole('spinbutton'), '20');
+
+      await user.click(screen.getByRole('button', { name: 'Generate' }));
+
+      expect(
+        screen.queryByText('Key expiration value needs to be greater than 0')
+      ).not.toBeInTheDocument();
+    });
+
+    it('should return on onGenerate the correct expiration date when months are selected with a valid quantity', async () => {
```

Comment:
I think you could wrap the next 3 tests in one with `it.each`, but it is fine they way it is as well